risc0/bigint2/src/ec/mod.rs

```diff
@@ -89,26 +88,22 @@ impl<const WIDTH: usize> AffinePoint<WIDTH> {
     /// Read the point from concatenated u32s for x and y
     ///
     /// Input interpreted as little-endian with x coordinate before y coordinate
-    pub fn from_u32s(
-        data: [[u32; WIDTH]; 2],
-        curve: &'static WeierstrassCurve<WIDTH>,
-    ) -> AffinePoint<WIDTH> {
-        AffinePoint {
-            buffer: data,
-            curve,
-        }
+    pub fn from_u32s(data: [[u32; WIDTH]; 2]) -> AffinePoint<WIDTH> {
+        AffinePoint { buffer: data }
     }
 }
 
 pub fn mul<const WIDTH: usize>(
     scalar: &[u32; WIDTH],
     point: &AffinePoint<WIDTH>,
-) -> AffinePoint<WIDTH> {
```

Comment:
Did you switch back to putting the result as a mutable parameter instead of as a return value for performance reasons?

Reply:
yes, and for consistency with RSA. Can always add higher-level APIs later for ergonomics?